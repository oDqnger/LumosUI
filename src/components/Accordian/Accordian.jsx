import PropTypes from 'prop-types'

export default function Accordian() {
    
}

Accordian.propTypes = {
    
}
