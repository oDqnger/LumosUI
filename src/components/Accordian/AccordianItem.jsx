import PropTypes from 'prop-types'

export default function AccordianItem() {

}

AccordianItem.propTypes = {
    
}