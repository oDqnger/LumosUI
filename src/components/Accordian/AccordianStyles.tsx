export const buttonStyles = "font-semibold mt-1 rounded-t-md text-left border-b-5 border-slate-700 w-96 h-14";
export const containerStyles = "p-2 rounded-b-md m-0 transition ease-in-out duration-300 border-2 border-zinc-500 w-96 inline-block overflow-auto opacity-100";
