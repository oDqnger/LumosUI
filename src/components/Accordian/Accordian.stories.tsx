import type { Meta, StoryObj } from "@storybook/react"
import Accordian from "./Accordian"
import AccordianItem from "./AccordianItem"
import "../../styles/output.css"

const meta: Meta<typeof Accordian> = {
    component: Accordian,
};

export default meta;

type Story = StoryObj<typeof Accordian>;

export const Something: Story = {
    args: {

    },
    render: () => 
    <>
        <Accordian>
            <AccordianItem 
            title="What are the job titles of a Frontend Developer">
                Front-end developers are also known as front-end engineers, front-end web developers, JavaScript Developers, HTML/CSS Developer, front-end web designers, and front-end web architects.

                Each of these roles mostly encompass the same front-end development skills but require different levels of expertise in different front-end development skills. It's better to look at the job description to get an idea about the job requirements.
            </AccordianItem>
            <AccordianItem title="How to become a Frontend Developer">
            Start with learning HTML and CSS; don't wait to fully master these and start building simple projects as soon as possible. You could try rebuilding the frontend of your favorite websites using HTML and CSS to start with. Do as many of these projects as possible as you keep learning. Once you are somewhat comfortable with HTML and CSS, start learning some basic JavaScript (DOM manipulation, making AJAX calls etc) and learn how to add interactivity to your websites. While you are at it learn some basics of Git and GitHub. At this point you should be able to get an entry level frontend development job. Keep revisiting this roadmap and try to fill the gaps in your knowledge.
            </AccordianItem>
        </Accordian>
    </>
}
