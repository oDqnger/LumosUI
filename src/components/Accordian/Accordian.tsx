import React, { Children, cloneElement } from 'react' 
import PropTypes from 'prop-types'
import AccordianItem from './AccordianItem'

export default function Accordian(props) {
    
    const {
        children: text,
    } = props;

    return (
        <>
            {
                Children.map(text, (child, index) => {
                    if (child.type === AccordianItem) {
                        return ( 
                            <>
                                {cloneElement(child, { key: index })}
                                <br /> 
                            </>
                        )
                    }
                })
            }
        </> 
    )
}

Accordian.propTypes = {
    
}
