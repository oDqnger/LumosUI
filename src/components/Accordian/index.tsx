import Accordian from './Accordian'
import AccordianItem from './AccordianItem'

export { Accordian, AccordianItem }
