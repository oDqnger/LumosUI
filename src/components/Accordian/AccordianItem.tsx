import React, { useState } from 'react'
import { Button } from "../Button"
import { containerStyles, buttonStyles } from './AccordianStyles';

export default function AccordianItem(props) {
    const {
        title,
        subtitle, 
        children: text,
        ...defaultProps
    } = props;

    const [displayContainer, setDisplayContainer] = useState(false);
    
    return (
        <>
            <Button
            className={buttonStyles}
            radius="none"
            onClick={(e) => {
                setDisplayContainer(!displayContainer)
            }}
            {...defaultProps}
            >{title}</Button>
            {
                displayContainer
                ?
                <>
                    <br/><div className={containerStyles}>
                        {text}
                    </div>
                </>
                :
                <></>
            }
        </>
 )
}