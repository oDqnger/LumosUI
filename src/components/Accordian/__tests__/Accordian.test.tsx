import '@testing-library/jest-dom/extend-expect';
import React from "react"
import { fireEvent, render } from "@testing-library/react"
import Accordian from "../Accordian"
import AccordianItem from '../AccordianItem';
import "@testing-library/jest-dom"

describe("Accordian component", () => {
    it("should render Accordian component", () => {
        const { container } = render(<Accordian></Accordian>)
        expect(container).toBeInTheDocument();
    })

    it("should match snapshot", () => {
        const { container } = render(<Accordian></Accordian>);
        expect(container).toMatchSnapshot();
    })

    it("should not allow anything other than AccordianItem component to render", () => {
        const { container: c1 } = render(
        <>
            <Accordian>Hello</Accordian>
        </>)
        expect(c1).toBeEmptyDOMElement()

        const { container: c2 } = render(
        <>
            <Accordian>
                <AccordianItem>Hello bro</AccordianItem>
            </Accordian>
        </>)
        expect(c1).toBeInTheDocument()
    })
})

describe("AccordianItem component", () => {
    it("should render Accordian component", () => {
        const { container } = render(<AccordianItem></AccordianItem>)
        expect(container).toBeInTheDocument();
    })

    it("should match snapshot", () => {
        const { container } = render(<AccordianItem></AccordianItem>);
        expect(container).toMatchSnapshot();
    })
})
