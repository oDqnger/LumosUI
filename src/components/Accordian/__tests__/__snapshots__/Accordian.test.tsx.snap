// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian component should match snapshot 1`] = `<div />`;

exports[`AccordianItem component should match snapshot 1`] = `
<div>
  <button
    class="mb-1 text-left border-b-5 border-slate-700 w-72 h-14border pr-7 pl-7 pt-2 pb-2 transition ease-in-out rounded-lg bg-zinc-500 hover:bg-zinc-600"
  />
</div>
`;
